lib/names/hostname.go

```diff
@@ -0,0 +1,15 @@
+package names
+
+import (
+	"os"
+	"strings"
+)
+
+// Hostname returns this hosts hostname, converting to lowercase so that
```

Comment:
I think the IPAM package should be update to use this too.  Although one niggle with that is that we need IPAM to work on v2 and v3 simultaneously (in order to implement the proposed CNI changes)

Reply:
suppose the v3 IPAM call should use this, v2 should continue as usual.